### PR TITLE
feat: stop web worker and prefetch and loading script via url parameter

### PR DESCRIPTION
**Pull request by akitaSummer**
hi，这次我添加了终止web worker和读取url信息的功能，您可以本地启动
http://localhost:3000/?auto_run=true&your_url=https%3A%2F%2Fgist.githubusercontent.com%2Flidangzzz%2F2adebd6a771f50e6430a34ec61d928af%2Fraw%2F1f9add39370b00dfbb3f5d20c8bc203dd09f52c2%2Fmy_first_notebook.hss
来进行查看

### CODE REVIEW

File: hedgehog-lab/src/HedgehogLab.tsx [hunk 1]
```diff
@@ -35,16 +36,17 @@ const HedgehogLab: React.FC<{}> = () => {
     onError: (lastError) => {
       // It's necessary to output all exception messages to user at output textbox,
       // including execution runtime exception and compiling exception -Lidang
-      console.log('Hedgehog Lab: Failed: ' + lastError.toString())
+      lastError ? console.log('Hedgehog Lab: Failed: ' + lastError.toString()) : console.log('Hedgehog Lab: Failed: web worker terminate')
       setResult({
```
Comment (CXuesong) [inline hedgehog-lab/src/HedgehogLab.tsx:40]: 建议在终止计算的时候给lastError传入一个非空值，否则这里的代码会变得有点绕……多一堆null check。

Comment (akitaSummer) [reply hedgehog-lab/src/HedgehogLab.tsx:40]: 当你中断的时候为空，当为空的时候，右侧results展示Please write your code on the left and click the 'Compile and run' button，而执行出错，右边会展示错误原因

Comment (akitaSummer) [reply hedgehog-lab/src/HedgehogLab.tsx:40]: console.log('Hedgehog Lab: Failed: ' + lastError.toString())
      setResult({
        outputItem: [],
        outputString: lastError.toString() === 'web worker terminate' ? '' : lastError.toString(),
      })
可能这样修改代码更易看懂？

Comment (CXuesong) [reply hedgehog-lab/src/HedgehogLab.tsx:40]: 哇……这样看起来更僵硬了😂

搞个AbortError之类的Error派生类如何？万一以后要使用程序判断是否是取消执行，只需要用instance of就够了。

Comment (akitaSummer) [reply hedgehog-lab/src/HedgehogLab.tsx:41]: console.log('Hedgehog Lab: Failed: ' + lastError.toString())
      setResult({
        outputItem: [],
        outputString: lastError instanceof AbortError ? '' : lastError.toString(),
      })
我已经做了修改，您可以看一下

File: hedgehog-lab/src/type/AbortError.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export default class AbortError extends Error {
+  constructor(str: string) {
+    super(str);
```
Comment (CXuesong) [inline hedgehog-lab/src/type/AbortError.ts:3]: 你需要在这里手动调用setPrototype，否则 instanceof Error 会返回 false，因为党党貌似用的是ES5编译模式

https://stackoverflow.com/questions/41102060/typescript-extending-error-class

Comment (akitaSummer) [reply hedgehog-lab/src/type/AbortError.ts:3]: 恩恩，学到了，谢谢😃

**Comment by AGDholo** [comment]
We have adjusted the core code, so conflicts need to be resolved.
The project is currently in a state of rapid iteration, so many things may be significantly updated. Thank you very much for your contribution, and we are deeply sorry for the creation of such conflicts

**Comment by imtsuki** [comment]
Sorry for the inconvenience, we just deleted the dev branch. Could you reopen a PR against the master branch?

**Pull request closed without merging**